src/uu/stdbuf/build.rs

```diff
@@ -19,31 +25,82 @@ mod platform {
     pub const DYLIB_EXT: &str = ".dylib";
 }
 
-#[cfg(target_os = "windows")]
-mod platform {
-    pub const DYLIB_EXT: &str = ".dll";
-}
-
 fn main() {
-    let current_exe = current_exe().unwrap();
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-changed=src/libstdbuf/src/libstdbuf.rs");
+
+    let out_dir = env::var("OUT_DIR").expect("OUT_DIR not set");
+    let target = env::var("TARGET").unwrap_or_else(|_| "unknown".to_string());
+
+    // Create a separate build directory for libstdbuf to avoid conflicts
+    let build_dir = Path::new(&out_dir).join("libstdbuf-build");
+    fs::create_dir_all(&build_dir).expect("Failed to create build directory");
+
+    // Get the cargo executable
+    let cargo = env::var("CARGO").unwrap_or_else(|_| "cargo".to_string());
+
+    // This manual cargo call ensures that libstdbuf is built before stdbuf.rs is compiled, which is necessary
+    // for include_bytes!(..."/libstdbuf.so") to work.
+    // In the future, "bindeps" should be used to simplify the code and avoid the manual cargo call,
+    // however this is available only in cargo nightly at the moment.
+    let mut cmd = Command::new(&cargo);
```

Comment:
i really don't like Command calls, they are to debug, can break dependencies analysis, etc :(
about bindeps, could you please add a link to the issue? thanks

Reply:
Yes, it is ugly, but I haven't found a more elegant solution :-/
I was surprised that cargo does not support something like bindeps yet. It seems quite a basic requirement to be able to say "build A before B".
I added a link to https://github.com/rust-lang/cargo/issues/9096 in the comment.

At least the good thing is that once bindeps is stabilized, the Command call can be removed, and something like this added to the Cargo.toml of stdbuf:
```
[dependencies]
libstdbuf = { ..., artifact = "cdylib" }
```